authentik/policies/geoip/models.py

```diff
@@ -87,6 +108,42 @@ def passes_country(self, request: PolicyRequest) -> PolicyResult:
 
         return PolicyResult(True)
 
+    def passes_distance(self, request: PolicyRequest) -> PolicyResult:
+        """Check if current policy execution is out of distance range compared
+        to previous authentication requests"""
+        # Get previous login event and GeoIP data
+        previous_logins = Event.objects.filter(
+            action=EventAction.LOGIN, user__pk=request.user.pk, context__geo__isnull=False
+        ).order_by("-created")[: self.history_login_count]
+        _now = now()
+        geoip_data: GeoIPDict | None = request.context.get("geoip")
+        if not geoip_data:
+            return PolicyResult(False)
+        for previous_login in previous_logins:
+            previous_login_geoip: GeoIPDict = previous_login.context.get("geo")
```

Comment:
```suggestion
            previous_login_geoip: GeoIPDict = previous_login.context["geo"]
```
We've checked previously. I'd rather it fails on that line if something went wrong, than a `None` error later on